src/NumericalAlgorithms/FiniteDifference/AoWeno.hpp

```diff
@@ -0,0 +1,334 @@
+ * u_{\xi2} &= -\frac{11}{336} u_{j-2} + \frac{53}{84} u_{j-1} -
+ *            \frac{67}{56} u_{j} + \frac{53}{84} u_{j+1} -
+ *            \frac{11}{336} u_{j+2} \\
+ * u_{\xi3} &= -\frac{1}{12} u_{j-2} + \frac{1}{6} u_{j-1}
+ *            -\frac{1}{6} u_{j+1} + \frac{1}{12} u_{j+2} \\
+ * u_{\xi4} &= \frac{1}{24} u_{j-2} -\frac{1}{6} u_{j-1}
+ *           +\frac{1}{4} u_{j} - \frac{1}{6}u_{j+1}+\frac{1}{24}u_{j+2}
+ * \f}
+ *
+ * The oscillation indicator is given by
+ *
+ * \f{align*}{
+ *  \beta^{r5}&=\left(u_{\xi}+\frac{1}{10}u_{\xi3}\right)^2 \\
+ *             &+\frac{13}{3}\left(u_{\xi2}+\frac{123}{455}u_{\xi4}\right)^2\\
+ *             &+\frac{781}{20}(u_{\xi3})^2+\frac{1421461}{2275}(u_{\xi4})^2
+ * \f}
+ *
+ * There are two linear weights \f$\gamma_{\mathrm{hi}}\f$ and
+ * \f$\gamma_{\mathrm{lo}}\f$. \f$\gamma_{\mathrm{hi}}\f$ controls how much of
+ * the 5th-order stencil is used in smooth regions, while
+ * \f$\gamma_{\mathrm{lo}}\f$ controls the linear weight of the central
+ * 3rd-order stencil. For larger \f$\gamma_{\mathrm{lo}}\f$, the 3rd-order
+ * method is more centrally weighted. The linear weights for the stencils
+ * are given by
+ *
+ * \f{align*}{
+ *  \gamma^{r5}&=\gamma_{\mathrm{hi}} \\
+ *  \gamma^{r3}_1& = (1-\gamma_{\mathrm{hi}})(1-\gamma_{\mathrm{lo}})/2 \\
+ *  \gamma^{r3}_1& = (1-\gamma_{\mathrm{hi}})\gamma_{\mathrm{lo}} \\
+ *  \gamma^{r3}_3& = (1-\gamma_{\mathrm{hi}})(1-\gamma_{\mathrm{lo}})/2
+ * \f}
+ *
+ * We use the standard nonlinear weights instead of the "Z" weights of
+ * \cite Borges20083191
+ *
+ * \f{align*}{
+ *  w^{r5}&=\frac{\gamma^{r5}}{(\beta^{r5}+\epsilon)^q} \\
+ *  w^{r3}_l&=\frac{\gamma^{r3}_l}{(\beta^{r3}_l+\epsilon)^q}
+ * \f}
+ *
+ * where \f$\epsilon\f$ is a small number used to avoid division by zero. The
+ * normalized nonlinear weights are denoted by \f$\bar{w}^{r5}\f$ and
+ * \f$\bar{w}_l^{r3}\f$. The final reconstructed polynomial \f$P(\xi)\f$ is
+ * given by
+ *
+ * \f{align*}{
+ *  P(\xi)&=\frac{\bar{w}^{r5}}{\gamma^{r5}}
+ *   \left(P^{r5}(\xi)-\gamma^{r3}_1P^{r3}_1(\xi)
+ *         -\gamma^{r3}_2P^{r3}_2(\xi)-\gamma^{r3}_2P^{r3}_2(\xi)\right) \\
+ *  &+\bar{w}^{r3}_1P^{r3}_1(\xi)+\bar{w}^{r3}_2P^{r3}_2(\xi)
+ *  +\bar{w}^{r3}_3P^{r3}_3(\xi)
+ * \f}
+ *
+ * The weights \f$\gamma_{\mathrm{hi}}\f$ and \f$\gamma_{\mathrm{lo}}\f$
+ * are typically chosen to be in the range \f$[0.85,0.95]\f$.
+ *
+ * ### First alternative oscillation indicators
+ *
+ * Instead of integrating over just the cell, we can instead integrate the basis
+ * functions over the entire fit interval, \f$[-5/2,5/2]\f$. Using this interval
+ * for the \f$S^{r3}_l\f$ and the \f$S^{r5}\f$ stencils we get
+ *
+ * \f{align*}{
+ *  \beta^{r3}_l&=(u_{\xi})^2 + \frac{37}{3} (u_{\xi2})^2 \\
+ *  \beta^{r5}  &=\left(u_{\xi}+\frac{61}{10}u_{\xi3}\right)^2
+ *   + \frac{569}{4}u_{\xi3}^2 \\
+ *  &+ \frac{1}{8190742}\left(5383 u_{\xi2} + 167158 u_{\xi4}\right)^2
+ *  +\frac{4410763}{501474}(u_{\xi4})^2
```

Comment:
This form is important because it very clearly shows that the oscillation indicator is always non-negative. I don't understand why you want me to change the last term.

Reply:
Ah, I see what you're saying now for the last term, thanks :)